docs/mathematical_and_physical_conventions.md

```diff
@@ -0,0 +1,179 @@
+
+```math
+A = A^i_\text{c} e_i = A^i b_i, 
+```
+
+Where the "c" subscript indicates that the considered components is computed in the Cartesian basis. It can be shown using the chain rule that the following equality holds
+
+```math
+A^i_\text{c} = J_{ij} A^j. 
+```
+
+Note that relating the components of the vector $`A`$ expressed in the covariant basis is less straightforward as this involves the metric tensor, i.e.  
+
+```math
+A^i_\text{c} = J_{ij} g^{jk}A_k. 
+```
+
+### Curvilinear to curvilinear
+
+Let us now consider the more general case where one seeks to relate two curvilinear coordinate system $`\{q^i\}`$ and $`\{p^i\}`$. We denote by $`J\{q\to p\}`$ the Jacobian of the transformation from the coordinate system $`\{q^i\}`$ to $`\{p^i\}`$. The components of the Jacobian and of its inverse are defined similarly as above as 
+
+```math
+J_{ij}\{q\to p\} = \frac{\partial p^i}{\partial q^j}, \quad J_{ij}^{-1}\{p\to q\} = \frac{\partial q^i}{\partial p^j}. 
+```
+
+Here one may note that $`J_{ij}^{-1}\{p\to q\} = J_{ij}\{q\to p\}`$. Let us now write $`\{b_i\}`$ (resp. $`\{b^i\}`$) the contravariant (resp. covariant) vector basis associated with coordinates $`\{q^i\}`$, and $`\{c_i\}`$ (resp. $`\{c^i\}`$) the contravariant (resp. covariant) vector basis associated with coordinates $`\{p^i\}`$. Let us now express a vector $`A`$ in these basis as 
+
+```math
+A = A^i\{p\} b_i = A_i\{p\} b^i = A^i\{q\} c_i = A_i\{q\} c^i, 
+```
+
+where $`A^i\{p\}`$ (resp. $`A_i\{p\}`$) refers to the $`i`$-th covariant (resp. contravariant) component of $`A`$ expressed in the vector basis associated with coordinates $`\{p^i\}`$, and similarly $`A^i\{q\}`$ (resp. $`A_i\{q\}`$) refers to the $`i`$-th covariant (resp. contravariant) component of $`A`$ expressed in the vector basis associated with coordinates $`\{q^i\}`$. We have
+
+```math
+A^i\{p\} = J_{ij}\{q\to p\} A^j\{p\}. 
+```
+
+By introducing the metric tensors $`G\{p\}`$  and $`G\{q\}`$ associated with both curvilinear coordinates system, and by writing $`g_{ij}\{p\}`$ and $`g_{ij}\{q\}`$ their elements ($`g^{ij}\{p\}`$ and $`g^{ij}\{q\}`$ the elements of their inverse) one relates the components of $`A`$ expressed in covariant vector basis associated with both curvilinear coordinates system as 
+
+```math
+g^{ik}\{p\}A_k\{p\} = J_{ij}\{q\to p\}g^{jl}\{p\} A_l\{p\}. 
+```
+
+## Differential operators
+
+Hereafter are expressed differential operators in a curvilinear coordinate system $`\{q^i\}`$. 
+
+### Gradient
+
+Let us consider a scalar field $`f`$. The gradient $`\nabla f`$ of such field is defined in the Cartesian basis $`\{e_i\}`$ as
+
+```math
+\nabla f = \frac{\partial f}{\partial x^i} e_i. 
+```
+
+This quantity can be expressed in the both contravariant $`\{b_i\}`$ and covariant $`\{b^i\}`$ vector basis as
+
+```math
+\nabla f = \frac{\partial f}{\partial q^i} b^i = g^{ij} \frac{\partial f}{\partial q^j} b_i. 
+```
+
```

Comment:
Maybe good to mention that it is much more common to see the definition in the contravariant basis